be/src/vec/functions/array/function_array_with_constant.cpp

```diff
@@ -91,9 +91,9 @@ class FunctionArrayWithConstant : public IFunction {
         array_sizes.reserve(input_rows_count);
         for (size_t i = 0; i < input_rows_count; ++i) {
             auto array_size = num->get_int(i);
-            if (UNLIKELY(array_size < 0)) {
-                return Status::RuntimeError("Array size can not be negative in function:" +
-                                            get_name());
+            if (UNLIKELY(array_size < 0) || UNLIKELY(array_size > max_array_size_as_field)) {
+                return Status::RuntimeError("Array size should in range(0, {}) in function: {}",
```

Comment:
add array_size actual value in error message for debug

Reply:
array_size has already in error message